.github/workflows/check-deep-tests-reusable.yml

```diff
@@ -19,4 +21,7 @@ jobs:
       with:
         script: |
           const script = require('${{ github.workspace }}/dafny/.github/workflows/check-for-workflow-run.js')
-          console.log(script({github, context, core}, workflowID = 'deep-tests.yml', branch = 'master', sha = '${{ inputs.sha }}'))
+          console.log(script({github, context, core,
+                              workflow_id: 'deep-tests.yml',
+                              ...('${{ inputs.sha }}' ? {sha: '${{ inputs.sha }}'} : {}),
+                              ...('${{ inputs.branch }}' ? {branch: '${{ inputs.branch }}'} : {})}))
```

Comment:
Cute! I suspect this isn't particularly idiomatic Javascript but not worth bikeshedding over. :)

Reply:
? It's completely standard, AFAIK.
You might argue that `('${{ inputs.branch }}' && {branch: '${{ inputs.branch }}'})` also complies with ES6, but it relies on corner cases.

Reply:
I meant it would be more idiomatic to pass the parameters unconditionally and check for empty strings on the callee side as needed.